docs: update docs with v1.0.0 changes (#47)
project-stacker/project-stacker.github.io: docs/whats-new.md

# What's New

## [v1.0.0](https://github.com/project-stacker/stacker/releases/tag/v1.0.0-rc9)

### Convert a Dockerfile for stacker

> - A new [`stacker convert`](reference/stacker_cli.md#stacker-convert) command performs a conversion of a Dockerfile into a stacker.yaml file. During the conversion, some variables from the Dockerfile may be exported to a substitution file.  

Reviewer: ```suggestion
- A new [`stacker convert`](reference/stacker_cli.md#stacker-convert) command performs a conversion of a Dockerfile into a stacker.yaml file. During the conversion, some variables from the Dockerfile may be exported to a substitution file to be used with the `--substitution-file` argument.  
```

Author: Done

    :pencil2: The conversion is a best-effort process and may not be successful in all cases.

### Publish specific layers

> - By default, the [`stacker publish`](reference/stacker_cli.md#stacker-publish) command pushes all layers in a stacker.yaml file instead of only the required layers. Using a new command option, `layer <value>`, you can explicitly specify which layers are to be published.  This command option can be specified multiple times, selecting each layer to be included.

Reviewer: i see that 'layer' is hard coded in the command arg. we should change that and change it here to say "image"

Author: Should I keep "layers" in this paragraph until the code is changed?

Reviewer: sure, that's fine, unless @rchincha has a reason to leave the option as `--layer`, in which case we'll need to explain it here.

also, when we do change it, let's add the dashes so it reads `--layer`

Author: When (if?) we change the argument name, should this section become "### Publish specific images"?  And should all instances of 'layer' be changed then to 'image'?

### Specify a single working directory

> - A new [`stacker`](reference/stacker_cli.md#stacker) command option, `work-dir`, sets the working directory for stacker's cache, OCI output, and rootfs output. The existing command options `stacker-dir`, `oci-dir`, and `roots-dir` can then be omitted or used to override the `work-dir` setting.

Reviewer: ```suggestion
- A new [`stacker`](reference/stacker_cli.md#stacker) command option, `--work-dir`, sets the working directory for stacker's cache, OCI output, and rootfs output. The existing command options `--stacker-dir`, `--oci-dir`, and `--roots-dir` can then be omitted or used to override the `--work-dir` setting.
```

Author: Done

> ### Import contents when no shell exists

Reviewer: ```suggestion
### Import contents when no shell exists in the base image
```

Author: Done

> - Import directives can include destination paths. This feature is useful when images are built from a scratch layer before any shell has been invoked. Prior to this, a `run:` section was required to invoke a shell and to explicitly copy files to be imported into the image. For example, you can now write a directive such as the following, with no `run:` section:

Reviewer: ```suggestion
- Import directives can include destination paths. This feature is useful to simplify `run` section scripts, and for when images are built from a scratch base image, which does not contain a shell to run the script in a `run` section. Prior to this, a `run:` section was required to invoke a shell and to explicitly copy files to be imported into the image. For example, you can now write a directive such as the following, with no `run:` section:
```

Reviewer: On a second reading, maybe the best way to explain this is "when images are built without a base image" instead of "a scratch base image"

@rchincha / @hallyn what do you guys think?

Author: Does this work?  "This feature is useful to simplify `run` section scripts, and for when images are built without a base image. With no base image, there is no shell to run the script in a `run` section."

Reviewer: sounds good to me, thanks

        test:
          from:
            type: scratch
          imports:
            - path: test_file
              dest: /files/
            - path: test_file2
              dest: /file2

### Generate SBOMs during the build

- Changes added in [OCI Distribution Spec v1.1.0-rc3](https://github.com/opencontainers/distribution-spec/releases/tag/v1.1.0-rc3) and [OCI Image Spec v1.1.0-rc4](https://github.com/opencontainers/image-spec/releases/tag/v1.1.0-rc4) (summarized [here](https://opencontainers.org/posts/blog/2023-07-07-summary-of-upcoming-changes-in-oci-image-and-distribution-specs-v-1-1/)) allow arbitrary artifact types and references. These changes support software supply chain use cases such as SBOMs.

- For a demonstration of an OCI artifacts workflow that generates an SBOM, see [Software Provenance Workflow Using OCI Artifacts](user_guide/generate_sbom.md).

### Report kernel version and fs type

- The [`stacker check`](reference/stacker_cli.md#stacker-check) command now reports this information.

### Build improvements

***

## [v0.40.1](https://github.com/project-stacker/stacker/releases/tag/v0.40.1)

### Support for `scratch`

> - Prior to v0.40.1, `stacker` did not support empty root filesystems to be used a base container image. The support has now been [added](reference/stacker_file.md#from) which can be used to host statically built binaries.

Reviewer: ```suggestion
- Prior to v0.40.1, `stacker` did not support empty root filesystems to be used as a base container image. The support has now been [added](reference/stacker_file.md#from) which can be used to host statically built binaries.
```

Author: Done

### Support for `import`ing content into container image

> - Prior to v0.40.1, copying content into a layer permanently involved bind mounting a shell such as busybox and invoking appropriate commands using the `run` directive. Now `import` directive [allows](reference/stacker_file.md#import-dest) for the `dest` option to achieve the same.

Reviewer: ```suggestion
- Prior to v0.40.1, copying content into an image permanently involved bind mounting a shell such as busybox and invoking appropriate commands using the `run` directive. Now `import` directive [allows](reference/stacker_file.md#import-dest) for the `dest` option to achieve the same.
```

Author: Done

> - Prior to v0.40.1, copying content into an image permanently involved bind mounting a shell such as busybox and invoking appropriate commands using the `run` directive. Now `import` directive [allows](reference/stacker_file.md#import-dest) for the `dest` option to achieve the same.

Reviewer: I missed this earlier, but this paragraph is only relevant in the context of scratch images. you could always copy into an image without bind mounting a shell if your base image had a shell already.

maybe this should read "prior to v0.40.1, copying content into a `scratch` image ..."
